test/EndToEndTests/Microsoft.OData.E2E.Portable/AsynchronousTests/AsynchronousQueryTests.cs

```diff
@@ -1197,6 +1197,36 @@ public void Linq_Where_GreaterThan()
             this.WaitForTestToComplete();
         }
 
+        /// <summary>
+        /// When DataServiceContext.MakeIdPredicateByKey=false, An expression with an Id as a predicate should create an Uri with a $filter in query options
+        /// </summary>
+        [Fact]
+        public void Linq_Where_Generates_Filter_By_Id_When_MakeIdPredicateByKey_Is_False()
+        {
+            var context = this.CreateWrappedContext<DefaultContainer>().Context;
+            context.MakeIdPredicateByKey = false;
+            var query = (from c in context.Customer
+                         where c.CustomerId == -10
+                         select new Customer { Name = c.Name, CustomerId = c.CustomerId }) as DataServiceQuery<Customer>;
+            var uri = query.ToString();
+            Assert.Contains("$filter=CustomerId eq -10", uri);

```

Comment:
I think you should also explicitly assert that it does not `Customers(-10)`. Or better yet, verify the entire url with an equality assertion (see related comments on the test below).

Reply:
Used `Assert.EndsWith`